### DISCUSSION TITLE
metabolic flux analysis/ elementary flux mode

### DISCUSSION THREAD

**Discussion by gcxxx** [opencobra/cobrapy#1247] [Q&A] [answered]
Hey guys,
I am trying to do some mfa and efm analysis. Is there a way to do this analysis by using cobrapy package?

**Comment by matthiaskoenig** [comment, marked as answer]
Not directly in cobrapy, but you could try `EFMlrs: a Python package for elementary flux mode enumeration via lexicographic reverse search`  based on cobrapy published recently:
https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-021-04417-9

> Background: Elementary flux mode (EFM) analysis is a well-established, yet computationally challenging approach to characterize metabolic networks. Standard algorithms require huge amounts of memory and lack scalability which limits their application to single servers and consequently limits a comprehensive analysis to medium-scale networks. Recently, Avis et al. developed mplrs-a parallel version of the lexicographic reverse search (lrs) algorithm, which, in principle, enables an EFM analysis on high-performance computing environments (Avis and Jordan. mplrs: a scalable parallel vertex/facet enumeration code. arXiv:1511.06487 , 2017). Here we test its applicability for EFM enumeration.
 
> Results: We developed EFMlrs, a Python package that gives users access to the enumeration capabilities of mplrs. EFMlrs uses COBRApy to process metabolic models from sbml files, performs loss-free compressions of the stoichiometric matrix, and generates suitable inputs for mplrs as well as efmtool, providing support not only for our proposed new method for EFM enumeration but also for already established tools. By leveraging COBRApy, EFMlrs also allows the application of additional reaction boundaries and seamlessly integrates into existing workflows.

> Conclusion: We show that due to mplrs's properties, the algorithm is perfectly suited for high-performance computing (HPC) and thus offers new possibilities for the unbiased analysis of substantially larger metabolic models via EFM analyses. EFMlrs is an open-source program that comes together with a designated workflow and can be easily installed via pip.

> Keywords: Cobrapy; Elementary modes; Lexicographic reverse search; Metabolic modelling; Mplrs; Systems biology. 

In addition there is also a python wrapper for the EFMTool which is often used for EFM analysis
https://csb.ethz.ch/tools/software/efmtool.html
https://pypi.org/project/efmtool/

**Reply by gcxxx** [reply]
Thank you for your answer! I will have a try😀
